opentreemap/works_management/models.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+from __future__ import unicode_literals
+from __future__ import division
+
+from django.contrib.gis.db import models
+from django.utils.translation import ugettext_lazy as _
+from django.utils import timezone
+
+from treemap.models import MapFeature, User
+from treemap.audit import Auditable
+from treemap.udf import UDFModel
+from treemap.instance import Instance
+
+
+class Team(models.Model):
+    instance = models.ForeignKey(Instance)
+    name = models.CharField(max_length=255)
+
+
+class WorkOrder(models.Model, Auditable):
+    instance = models.ForeignKey(Instance)
+    name = models.CharField(max_length=255)
+    created_at = models.DateTimeField(auto_now_add=True)
+    created_by = models.ForeignKey(User)
+    updated_at = models.DateTimeField(auto_now=True)
+
+
+class Task(UDFModel, Auditable):
+    REQUESTED = 0
+    SCHEDULED = 1
+    COMPLETED = 2
+    CANCELED = 3
+
+    STATUS_CHOICES = (
+        (REQUESTED, _('Requested')),
+        (SCHEDULED, _('Scheduled')),
+        (COMPLETED, _('Completed')),
+        (CANCELED, _('Canceled')),
+    )
+
+    instance = models.ForeignKey(Instance)
+    map_feature = models.ForeignKey(MapFeature)
+    work_order = models.ForeignKey(WorkOrder, null=True, related_name='tasks')
+    team = models.ForeignKey(Team, null=True)
+
+    name = models.CharField(max_length=255)
+    office_notes = models.TextField()
+    field_notes = models.TextField()
+
+    status = models.IntegerField(
+        choices=STATUS_CHOICES,
+        default=REQUESTED)
+
+    requested_on = models.DateField()
+    scheduled_on = models.DateField()
+    closed_on = models.DateField()
+
+    created_at = models.DateTimeField(auto_now_add=True)
+    created_by = models.ForeignKey(User)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    udf_settings = {
+        'Action': {
+            'iscollection': False,
+            'is_protected': True,
+            'defaults': {
+                'type': 'choice',
+                'protected_choices': [
+                    'Plant Tree',
+                    'Remove Tree',
+                ],
+            }
+        },
+    }
+
+    def save_with_user(self, user, *args, **kwargs):
+        """
+        Update WorkOrder fields when Task is saved.
+        """
+        if self.work_order:
+            self.work_order.updated_at = timezone.now()
+            self.work_order.save()
```

Comment:
I wonder why this doesn't trigger an exception in the unit test? I'll investigate.